### PR TITLE
re2: 20201001 -> 20210401

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119742` at 487a99b8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>45 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.sysdig</li>
    <li>linuxPackages_4_4.sysdig</li>
    <li>linuxPackages_4_9.sysdig</li>
    <li>linuxPackages_hardkernel_4_14.sysdig</li>
    <li>linuxPackages_hardkernel_latest.sysdig</li>
    <li>linuxPackages_latest-libre.sysdig</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>150 packages skipped due to time constraints:</summary>
  <ul>
    <li>arrow-cpp</li>
    <li>beancount (python38Packages.beancount)</li>
    <li>bear</li>
    <li>chromium</li>
    <li>clickhouse</li>
    <li>deja-dup</li>
    <li>dnsdist</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>dvc-with-remotes</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>cre2</li>
    <li>grpc</li>
    <li>python38Packages.fb-re2</li>
    <li>python38Packages.googleapis-common-protos</li>
    <li>python39Packages.fb-re2</li>
    <li>python39Packages.googleapis-common-protos</li>
    <li>re2</li>
    <li>rubyPackages.re2 (rubyPackages_2_6.re2)</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/re2/default.nix:35:3:
```
   |
35 |   meta = {
   |   ^
```
</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Review by lukegb — APPROVED** [review]

File: pkgs/development/libraries/re2/default.nix [hunk 1]
```diff
@@ -1,14 +1,18 @@
 { lib, stdenv, fetchFromGitHub }:
 
+let
+  rev = "2021-04-01";
+in
+
 stdenv.mkDerivation {
   pname = "re2";
-  version = "20201001";
+  version = builtins.replaceStrings [ "-" ] [ "" ] rev;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/re2/default.nix:9]:
```suggestion
  version = "2021-04-01";
```

File: pkgs/development/libraries/re2/default.nix [hunk 2]
```diff
@@ -1,14 +1,18 @@
 { lib, stdenv, fetchFromGitHub }:
 
+let
+  rev = "2021-04-01";
+in
+
 stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/re2/default.nix:7]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/re2/default.nix [hunk 3]
```diff
@@ -1,14 +1,18 @@
 { lib, stdenv, fetchFromGitHub }:
 
+let
+  rev = "2021-04-01";
+in
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/re2/default.nix:6]:
```suggestion
```

File: pkgs/development/libraries/re2/default.nix [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
   pname = "re2";
-  version = "20201001";
+  version = "2021-04-01";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "re2";
-    rev = "2020-10-01";
-    sha256 = "0a5f7av1pk6p3jxc2w6prl00lyrplap97m68hnhw7jllnwljk0bx";
+    inherit rev;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/re2/default.nix:10]:
```suggestion
    rev = version;
```

**Pull request merged**